validator/sawtooth_validator/execution/processor_handlers.py

```diff
@@ -83,3 +83,41 @@ def handle(self, connection_id, message_content):
             status=HandlerStatus.RETURN,
             message_out=ack,
             message_type=validator_pb2.Message.TP_UNREGISTER_RESPONSE)
+
+
+class ProcessorAddReceiptDataHandler(Handler):
+    def __init__(self, processor_collection):
+        self._collection = processor_collection
```

Comment:
These should be in `validator/execution/tp_state_handlers.py` and resemble those handlers. They need to take a context manager as an argument in the constructor as that is where events and receipt data will be added to contexts.